src/main/scala/com.snowplowanalytics.sauna/observers/AzureEventHubsObserver.scala

```diff
@@ -0,0 +1,139 @@
+package com.snowplowanalytics.sauna
+package observers
+
+// java
+import java.util.Date
+import java.util.function.Consumer
+import java.util.concurrent.ExecutionException
+
+
+// scala
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.language.postfixOps
+import scala.collection.JavaConverters._
+import scala.util.{Try, Failure, Success}
+
+// akka
+import akka.actor._
+
+// eventhubs
+import com.microsoft.azure.eventprocessorhost.ExceptionReceivedEventArgs
+import com.microsoft.azure.eventhubs.EventData
+import com.microsoft.azure.eventprocessorhost.CloseReason
+import com.microsoft.azure.eventprocessorhost.IEventProcessor
+import com.microsoft.azure.eventprocessorhost.PartitionContext
+import com.microsoft.azure.eventprocessorhost._
+import com.microsoft.azure.servicebus.ConnectionStringBuilder
+import com.microsoft.azure.eventhubs.EventData
+
+// sauna
+import Observer.EventHubRecordReceived
+
+class AzureEventHubsObserver( val eventHubName: String, val namespaceName: String,
+	val eventHubConnectionString: String,
+	val storageConnectionString: String,
+	val consumerGroupName: Option[String]) extends Actor with Observer {
+	import AzureEventHubsObserver._
+
+	//val consumerGroupName = 
+	//val namespaceName = "saunadev"
+	//val eventHubName = "saunadev"
+	//val sasKeyName = "-----SharedAccessSignatureKeyName-----"
+	//val sasKey = "---SharedAccessSignatureKey----"
+	
+	//val storageAccountName = "saunadev"
+	//val storageAccountKey = "---StorageAccountKey----"
+	//val eventHubConnectionString = new ConnectionStringBuilder(namespaceName, eventHubName, sasKeyName, sasKey)
+	//val evhubConStr = "Endpoint=sb://saunadev.servicebus.windows.net/;SharedAccessKeyName=Policy;SharedAccessKey=6a1g8rNAxbJqh1dL6A//3/wXfX1pS9ZgpxVopahCgQE=;EntityPath=saunadev"
+	//val storageConStr = "DefaultEndpointsProtocol=https;AccountName=saunadev;AccountKey=L/mJMUSJSlfSsB3zQw87iVfj1Sj3hvqW5BgiHyoLZqXbyMpP+t7+tODZQKi0JJWVVwPmrb5w30ZL5tUnEGekNQ==;EndpointSuffix=core.windows.net" 
+	val host: EventProcessorHost  = new EventProcessorHost(eventHubName, consumerGroupName.getOrElse("$Default"), eventHubConnectionString, storageConnectionString)
+	notify("Registering host named " + host.getHostName())
+	val options: EventProcessorOptions = new EventProcessorOptions()
+	options.setExceptionNotification(new ErrorNotificationHandler())
+
+	val notifier = (s: String) => notify(s)
+	
+	override def preStart: Unit = {
+		try{		
+			host.registerEventProcessorFactory(new EventProcessorFactory(context, notifier), options).get()
+		}catch {
+			case e: Exception => 
+				notify(s"Failure while registering: ${e.toString()}")
+			case e: ExecutionException => 
+				notify(e.getCause().toString())
```

Comment:
this is most likely dead code since `ExecutionException` surely is a subtype of `Exception`